### DISCUSSION TITLE
Accessing a non-linear variable from a contour to compute the residual term for integrated boundary condition

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#26352] [Q&A General]
Hello everyone, 

In order to be able to impose a dynamic wetting boundary condition, first I need to define the c=0 contour (c is the order parameter). After that, I need to access the velocity on the closest point to the c=0 contour to the wall! Because at wall, the no-slip velocity boundary condition is used. So I need the velocity on the second closet point on the wall. Is this possible in MOOSE kernel system?

If this is not possible, is it possible to for example define a some y-intercept of the c=0 contour and find the velocity there?

Thanks,
Ali

**Comment by GiudGiud** [comment]
Hello

If you need this to work efficiently in parallel (MPI), it s going to be quite a bit of work. You ll need to create a new relationship manager than introduces ghosting of variable values from a contour to the wall / this point near the wall.

If not (threading is fine, or inefficient global comm of the contour), I would make a user object that:
- detects which elements are on the c=0 contour
- fills a KDTree with those elements
Then the BC can query the user object to:
- get the KDTRee from the UO
- query the KDTree for the closest distance from the wall to the contour

To work on the 1st and 2nd nodes near the wall, I would make a nodeset, with something close to a boundary layer generator. We dont have that one in MOOSE yet either, but we could use one.

Guillaume
